src/uct/ib/rc/base/rc_iface.c

```diff
@@ -935,45 +935,50 @@ ucs_status_t uct_rc_iface_qp_connect(uct_rc_iface_t *iface, struct ibv_qp *qp,
     return UCS_OK;
 }
 
-int uct_rc_iface_should_poll_rx(uct_rc_iface_t *iface, unsigned events,
-                                int *solicited)
+void uct_rc_iface_should_arm_cq(uct_rc_iface_t *iface, unsigned events,
+                                uint64_t *dirs_p, int *solicited_p)
 {
     int arm_rx_solicited, arm_rx_all;
+    uint64_t dirs;
 
     arm_rx_solicited = 0;
     arm_rx_all       = 0;
+    dirs             = 0;
     if (events & UCT_EVENT_RECV) {
         arm_rx_solicited = 1; /* to wake up on active messages */
     }
     if ((events & UCT_EVENT_SEND_COMP) && iface->config.fc_enabled) {
         arm_rx_all = 1; /* to wake up on FC grants (or if forced) */
     }
 
-    *solicited = arm_rx_solicited && !arm_rx_all;
-    return arm_rx_solicited || arm_rx_all;
+    if (events & UCT_EVENT_SEND_COMP) {
+        dirs |= UCS_BIT(UCT_IB_DIR_TX);
+    }
+
+    if (arm_rx_solicited || arm_rx_all) {
+        dirs |= UCS_BIT(UCT_IB_DIR_RX);
+    }
+
+    solicited_p[UCT_IB_DIR_TX] = 0;
+    solicited_p[UCT_IB_DIR_RX] = arm_rx_solicited && !arm_rx_all;
+    *dirs_p                    = dirs;
```

Comment:
IMO can set *dirs_p directly, no need helper var "dirs" - since this func returns void and has no error flow